actors/builtin/miner/miner_actor.go

```diff
@@ -520,13 +539,11 @@ func (a Actor) PreCommitSector(rt Runtime, params *SectorPreCommitInfo) *adt.Emp
 			rt.Abortf(exitcode.ErrIllegalState, "sector %v already committed", params.SectorNumber)
 		}
 
-		// Require sector lifetime meets minimum by assuming activation happens at last epoch permitted for seal proof.
-		// This could make sector maximum lifetime validation more lenient if the maximum sector limit isn't hit first.
-		maxActivation := rt.CurrEpoch() + MaxSealDuration[params.SealProof]
-		validateExpiration(rt, maxActivation, params.Expiration, params.SealProof)
-
 		depositMinimum := big.Zero()
 		if params.ReplaceCapacity {
+			// NOTE: we don't validate that the replacement sector
+			// lives at the target deadline/partition. If it
+			// doesn't, sealing may still succeed but replacement will fail.
```

Comment:
This is validated within `validateReplaceSector` -> `CheckSectorHealth`

Reply:
And it probably should be validated if it isn't, because you might be able to queue up a replacement for a sector that hasn't been proven yet.

Reply:
Hm. You're right. Although the target sector could expire before we even finish... but I guess that doesn't matter.

> because you might be able to queue up a replacement for a sector that hasn't been proven yet.

We _do_ check to make sure that the sector to-be-replaced has been proven.